larreco/RecoAlg/TrackMomentumCalculator.cxx

```diff
@@ -275,38 +453,64 @@ namespace trkf {
     auto const seg_steps = segments->x.size();
     if (seg_steps < 2) return -1;
 
-    double const recoL = segments->L.at(seg_steps - 1);
-    if (recoL < minLength || recoL > maxLength) return -1;
+    double const recoSegmentLength = segments->L.at(seg_steps - 1);
+    if (recoSegmentLength < minLength || recoSegmentLength > maxLength) return -1;
 
-    std::vector<float> dEi;
-    std::vector<float> dEj;
-    std::vector<float> dthij;
-    std::vector<float> ind;
-    if (getDeltaThetaij_(dEi, dEj, dthij, ind, *segments, seg_size) != 0) return -1.0;
+    std::vector<double> dEi;
+    std::vector<double> dEj;
+    std::vector<double> dthij;
+    std::vector<double> ind;
+    std::vector<bool> dthij_valid = segments->nvalid;
+    if (getDeltaThetaij_(dEi, dEj, dthij, ind, *segments, seg_size) != 0) return -1;
 
-    double logL = 1e+16;
-    double bf = -666.0; // double errs = -666.0;
+    auto const ndEi = dEi.size();
+    if (ndEi < 1) return -1;
 
-    int const start1{};
-    int const end1{static_cast<int>(maxMomentum_MeV / MomentumStep_MeV)};
-    int const start2{};
-    int const end2{max_resolution}; // 800.0;
+    double correction = 1.;
+    if (fMCSAngleMethod == kAngleCombined) { correction = std::sqrt(2.); }
 
-    for (int k = start1; k <= end1; ++k) {
-      double const p_test = 0.001 + k * 0.01;
+    // Gets energy evaluated by range
+    auto recoL = trk->Length();
+    double const minP = this->GetTrackMomentum(recoL, 13);
 
-      for (int l = start2; l <= end2; ++l) {
-        double const res_test = (start2 == end2) ? 2.0 : 0.001 + l * 1.0; // 0.001+l*1.0;
-        double const fv = my_mcs_llhd(dEi, dEj, dthij, ind, p_test, res_test);
+    ROOT::Minuit2::Minuit2Minimizer mP{};
+    FcnWrapperLLHD const wrapper{
+      (dEi), (dEj), (dthij), (ind), (dthij_valid), (seg_size), (correction)};
```

Comment:
All of the containers can be moved to the `FcnWrapperLLHD` constructor:

```c++
FcnWrapperLLHD const wrapper{std::move(dEi), 
                             std::move(dEj), 
                             std::move(dthij), 
                             std::move(ind), 
                             std::move(dthij_valid), 
                             seg_size, 
                             correction};
```

But see change to the `FcnWrapperLLHD` constructor.